test/functionalTest/cases/2176_not_print_spurious_decimals/not_print_spurious_decimals.test

```diff
@@ -0,0 +1,138 @@
+# Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Avoid printing spurious decimals in attribute values
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# 01. Create entity with attributes A: 1.123456789123456789 and B: 200
+# 02. Get A value: 1.123456789 (9 decimals)
+# 03. Get B value: 200 (0 decimals)
+# 04. Get whole entity, same results for A and B
+#
+
+
+echo "01. Create entity with attributes A: 1.123456789123456789 and B: 200"
+echo "===================================================================="
+payload='{
+  "id": "E",
+  "type": "T",
+  "A": {
+    "value": 1.123456789123456789,
+    "type": "Number"
+  },
+  "B": {
+    "value": 200,
+    "type": "Number"
+  }
+
+}'
+orionCurl --url '/v2/entities' --payload "$payload"
+echo
+echo
+
+
+echo "02. Get A value: 1.123456789 (9 decimals)"
+echo "========================================="
+orionCurl --url '/v2/entities/E/attrs/A/value' --out '*/*'
+echo
+echo
+
+
+echo "03. Get B value: 200 (0 decimals)"
+echo "================================="
+orionCurl --url '/v2/entities/E/attrs/B/value' --out '*/*'
+echo
+echo
+
+
+echo "04. Get whole entity, same results for A and B"
+echo "=============================================="
+orionCurl --url '/v2/entities/E/attrs'
+echo
+echo
+
+
+--REGEXPECT--
+01. Create entity with attributes A: 1.123456789123456789 and B: 200
+====================================================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/entities/E?type=T
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+02. Get A value: 1.123456789 (9 decimals)
+=========================================
+HTTP/1.1 200 OK
+Content-Length: 10
+Content-Type: text/plain
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+1.12345679
```

Comment:
Contrary to intuition, default behaviour is that `std::setprecision()` digits include both decimal and integer parts. This can be changed using, `std::fixed`. With that modifier, `std::setprecision()` sets the decimal part length... but the side-effect of using `std::fixed` is that float without decimal part are printed with the `.0`.

I'll try to fix this specializing the template for float and checking if the float have decimal part or not before applying modifiers.